### ISSUE TITLE
33 million normal cells not match CELLxGENE census version 15 May 2023

### ISSUE THREAD

**Issue by litxiaoyao** [bowang-lab/scGPT#239]
There are 33 million cells all in CELLxGENE census version 15 May 2023, but only ~22M normal.
![image](https://github.com/user-attachments/assets/5ec611b7-e2f9-4a0b-98a3-253dc028e2bd)
So is it wrong described in the paper?
![image](https://github.com/user-attachments/assets/a365d39c-2116-4779-977e-45901e984fee)

**Comment by litxiaoyao** [comment]
Besides, only with these 22M normal cells can end up to 51 tissues mentioned in the paper.
![7d6d00a0f76d5aeba8ba69163e5d382](https://github.com/user-attachments/assets/18a41c3e-4132-4243-8fc6-bad91849a048)

![image](https://github.com/user-attachments/assets/126afbb5-c465-4751-a6a7-22352d554781)

**Comment by yubin-ai** [comment]
can you check that what these numbers look like after filtering for is_primary_data in adata.obs?

**Comment by litxiaoyao** [comment]
All previous numbers are filtered with is_primary_data == True.

**Comment by subercui** [comment]
Hi @litxiaoyao , we didn't filetered by is_primary_data when we trained the model. You can find the code for downloading data in the data forlder. The reason was just we were not sure the exact meaning of what this is_primary_data means around the time, and we didn't find whether that means the cell is a duplicate or not. So we decided to train with all noraml cells by a straight-forward pool from cellxgene cencus, the training set we eventually have was a bit over 33 million.

**Comment by yubin-ai** [comment]
@subercui @litxiaoyao yes, I recently found that scGPT did not filter for that and it resulted in duplicates of cells. the difference in cell number is a result of that. 
I was about to open an issue and mention this but glad it is already discussed here.

**Comment by litxiaoyao** [comment]
@subercui , Hi, however, there are ~37M healthy cells if don't set is_primary_data == True. By the way, the data folder will download only 7 tissue_organs by default.
![image](https://github.com/user-attachments/assets/d901970d-098f-40ef-bf9e-d0e5402e8a52)

**Comment by litxiaoyao** [comment]
The index files produced by the data folder can also gather 37280217 cells in total.
![image](https://github.com/user-attachments/assets/1fb893d5-49f2-4b10-8575-6f5f9e93d99e)

**Comment by litxiaoyao** [comment]
Will this difference caused by the version of CxG between 15 May 2023 and 8 May 2023? So the real version is not 15 May 2023 but 8 May 2023 stated in the paper?
![image](https://github.com/user-attachments/assets/cbe15863-218e-4c53-a0af-d689d3ffd35e)
